nvflare/fuel/utils/config.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Dict, List, Optional
+
+
+class ConfigFormat(Enum):
+    JSON = "JSON"
+    PYHOCON = "PYHOCON"
+    OMEGACONF = "OMEGACONF"
+
+    @classmethod
+    def _config_format_extensions(cls):
+        return {
+            "JSON": [".json", ".json.default"],
+            "PYHOCON": [".conf", ".conf.default"],
+            "OMEGACONF": [".yml", ".yml.default"],
+        }
+
+    @classmethod
+    def search_order(cls):
+        return [cls.JSON, cls.PYHOCON, cls.OMEGACONF]
+
+    @classmethod
+    def get_extensions(cls, fmt: str) -> Optional[List[str]]:
+        return cls._config_format_extensions().get(fmt, None)
+
+    @classmethod
+    def config_exts(cls) -> str:
+        search_sequence: List[ConfigFormat] = cls.search_order()
+        extensions = []
+        for fmt in search_sequence:
+            exts = cls.get_extensions(fmt.name)
+            if exts:
+                extensions.extend(exts)
+
+        return "|".join(iter(extensions))
```

Comment:
Why joining into a str, instead of returning the list?

Reply:
You could also directly use the list in your error message. It will print quite nicely as a list.